genum-derive/tests/test.rs

```diff
@@ -0,0 +1,55 @@
+use ::genum_derive::GEnum;
+use glib::prelude::*;
+use glib::translate::{FromGlib, ToGlib};
+
+#[test]
+fn derive_genum() {
+    #[derive(Debug, Eq, PartialEq, Clone, Copy, GEnum)]
+    #[repr(u32)]
```

Comment:
Should the `repr` attribute maybe be added by the derive macro directly?

Reply:
Do we really need it actually? It seems to work fine without it.

Reply:
This way you make sure that it becomes a compiler error if the enum actually contains items that can't be represented as `u32`

Reply:
I don't know how to tweak the macro to generate this before the `enum {}` block and the rest after it. Any idea?

Reply:
I don't, so maybe let's keep that for a later time. Your `FromGlib` / `ToGlib` impls would fail to compile if the enum is of an incompatible shape so it should be fine.